uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/controllers/sub-controllers/leave-request-ctrl_test.js

```diff
@@ -1197,40 +1198,56 @@
         });
 
         describe('after contact is selected', function () {
-          var approvalStatus;
+          describe('when entitlement is present', function() {
+            var approvalStatus;
 
-          beforeEach(function () {
-            approvalStatus = optionGroupMock.specificValue('hrleaveandabsences_leave_request_status', 'value', '1');
-            $ctrl.initAfterContactSelection();
-            $scope.$digest();
-          });
+            beforeEach(function () {
+              approvalStatus = optionGroupMock.specificValue('hrleaveandabsences_leave_request_status', 'value', '1');
+              $ctrl.initAfterContactSelection();
+              $scope.$digest();
+            });
 
-          it('sets manager role', function () {
-            expect($ctrl.isRole('manager')).toBeTruthy();
-          });
+            it('sets manager role', function () {
+              expect($ctrl.isRole('manager')).toBeTruthy();
+            });
 
-          it('sets create mode', function () {
-            expect($ctrl.isMode('create')).toBeTruthy();
-          });
+            it('sets create mode', function () {
+              expect($ctrl.isMode('create')).toBeTruthy();
+            });
 
-          it('does not initialize absence types', function () {
-            expect(AbsenceTypeAPI.all).toHaveBeenCalled();
-          });
+            it('does not initialize absence types', function () {
+              expect(AbsenceTypeAPI.all).toHaveBeenCalled();
+            });
 
-          it('sets status to approved', function () {
-            expect($ctrl.request.status_id).toEqual(approvalStatus);
-          });
+            it('sets status to approved', function () {
+              expect($ctrl.request.status_id).toEqual(approvalStatus);
+            });
 
-          describe('cancelled status', function () {
-            var cancelStatus, availableStatuses;
+            describe('cancelled status', function () {
+              var cancelStatus, availableStatuses;
 
-            beforeEach(function () {
-              cancelStatus = optionGroupMock.specificObject('hrleaveandabsences_leave_request_status', 'name', 'cancelled');
-              availableStatuses = $ctrl.getStatuses();
+              beforeEach(function () {
+                cancelStatus = optionGroupMock.specificObject('hrleaveandabsences_leave_request_status', 'name', 'cancelled');
+                availableStatuses = $ctrl.getStatuses();
+              });
+
+              it('is not available', function () {
+                expect(availableStatuses).not.toContain(cancelStatus);
+              });
+            });
+          });
+
+          describe('when entitlement is not present', function() {
+            beforeEach(function() {
+              spyOn(Entitlement, 'all').and.callFake(function () {
```

Comment:
can the spyOn be on EntitlementAPI itself and avoid adding Entitlement in this file?